service/src/main/java/com/inventory/services/EmployeeServiceImpl.java

```diff
@@ -41,24 +42,49 @@ public Employee login(String email, String password) {
             return null;
     }
 
-
     @Override
     @Transactional
-    public List<Employee> getSuperiorList(Paging paging) {
-        List<Employee> list = employeeRepository.findAll();
+    public List<Employee> getSuperiorList(String name, Paging paging) {
         List<Employee> listOfSuperior = new ArrayList<>();
-        for (Employee employee : list) {
-            if (employee.getSuperiorId().equals("null")) {
-                listOfSuperior.add(employee);
-            }
+        List<Employee> listOfEmployee = getEmployeeListFromRepository(name, paging);
+        for (Employee em : listOfEmployee) {
+            if (em.getSuperiorId().matches("null"))
+                listOfSuperior.add(em);
+        }
+        List<Employee> listOfSortedSuperior = setSortedDataWithPaging(paging, listOfSuperior);
+        return listOfSortedSuperior;
+    }
+
+    private List<Employee> setSortedDataWithPaging(Paging paging, List<Employee> listOfEmployee) {
```

Comment:
'paging' should be passed to repository, so the results from repository are already paged